src/java.base/share/classes/java/nio/channels/Channels.java

```diff
@@ -532,7 +535,7 @@ public static Reader newReader(ReadableByteChannel ch, Charset charset) {
      * @param  minBufferCap
      *         The minimum capacity of the internal byte buffer,
      *         or {@code -1} if an implementation-dependent
-     *         default capacity is to be used
+     *         default capacity is to be used; may be ignored
```

Comment:
I'm not understanding this. The implNote says it *is* ignored, the annotation on the \@param says *may*.

```
     *         The minimum capacity of the internal byte buffer,
     *         or {@code -1} if an implementation-dependent default capacity is to be used.
     *         The minimum capacity is ignored.
```

Reply:
> I'm not understanding this. The implNote says it _is_ ignored, the annotation on the @param says _may_.

The spec allows the value to be ignored, the JDK implementation ignores the value. So it's spec vs. implementation.